x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrRetentionLeases.java

```diff
@@ -79,7 +81,10 @@ public static Optional<RetentionLeaseAlreadyExistsException> syncAddRetentionLea
         final TimeValue timeout
     ) {
         try {
-            final PlainActionFuture<ActionResponse.Empty> response = new PlainActionFuture<>();
+            final PlainActionFuture<ActionResponse.Empty> response = new UnsafePlainActionFuture<>(
+                Ccr.CCR_THREAD_POOL_NAME,
+                ThreadPool.Names.GENERIC
```

Comment:
I believe we need only GENERIC here?

```suggestion
                ThreadPool.Names.GENERIC
```

Reply:
Looking closely these names are for the waiter thread. So you are right. Only `generic` is needed. See b11a09ac957c88a1bc4fd193278c2ca8be7ab7a6